data/vibe/data/json.d

```diff
@@ -2188,6 +2190,18 @@ struct JsonStringSerializer(R, bool pretty = false)
 	}
 }
 
+unittest {
+	static assert(doesSerializerSupportStringSink!(JsonStringSerializer!(Appender!string)));
+
+	auto app = appender!string;
+	auto ser = JsonStringSerializer!(Appender!string)(app);
+	static struct T1 { void toString(scope void delegate(scope const(char)[])) {} }
+	static struct T2 { void toString(R)(scope ref R dst) { dst.put('f'); dst.put("foo"); } }
```

Comment:
Okay now I understand better what you meant. I don't really see a need for the second overload myself, as the `delegate` is a superset of this. It's trivial to wrap an output range in a delegate and does not allocate.

Reply:
True, but I think it makes sense to try to be compatible with [`formatValue`](https://dlang.org/library/std/format/format_value.html) in that regard, modulo the `fmt` parameter, since ranges are more idiomatic today and such an overload may already be defined for non-serialization purposes.